crates/solver/src/liquidity_collector.rs

```diff
@@ -43,3 +45,136 @@ impl LiquidityCollecting for LiquidityCollector {
         Ok(amms)
     }
 }
+
+/// A liquidity source which might not be initialised on creation. Instead
+/// initialisation gets retried in a background task over and over until it
+/// succeeds. Until the liquidity source has been initialised no liquidity will
+/// be provided.
+pub struct BackgroundInitLiquiditySource<L> {
+    liquidity_source: Arc<Mutex<Option<L>>>,
+}
+
+impl<L> BackgroundInitLiquiditySource<L> {
+    /// Creates a new liquidity source which might only be initialized at a
+    /// later point in time.
+    pub fn new<I, F>(label: &str, init: I, retry_init_timeout: Duration) -> Self
+    where
+        I: Fn() -> F + Send + Sync + 'static,
+        F: Future<Output = Result<L>> + Send,
+        L: LiquidityCollecting + 'static,
+    {
+        let liquidity_source: Arc<Mutex<Option<L>>> = Default::default();
+        let inner = liquidity_source.clone();
+        tokio::task::spawn(
+            async move {
+                loop {
+                    match init().await {
+                        Err(err) => {
+                            tracing::warn!(
+                                "failed to initialise liquidity source; next init attempt in \
+                                 {retry_init_timeout:?}: {err:?}"
```

Comment:
nit: `Duration: Debug` is kind of ugly - I would do something like:
```suggestion
                                 {}s: {err:?}", retry_init_timeout.as_secs_f64()
```

Reply:
That will literally print the same thing. 😅

Reply:
What? I thought `Duration: Debug` prints something like `Duration { secs: 0, nanosecs: 1000 }` - or am I thinking of a different thing?

Reply:
That's `Instant`

Reply:
Nice! I'm in the [lucky 10000](https://xkcd.com/1053/) today!